src/promptflow/promptflow/_sdk/_service/entry.py

```diff
@@ -1,16 +1,45 @@
 # ---------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # ---------------------------------------------------------
+import argparse
+import stat
+import sys
 
 import waitress
+import yaml
 
-from promptflow._sdk._constants import LOCAL_SERVICE_PORT
+from promptflow._sdk._constants import HOME_PROMPT_FLOW_DIR, SERVICE_CONFIG_FILE
 from promptflow._sdk._service.app import create_app
+from promptflow._sdk._service.utils import get_random_port, is_port_in_use
+from promptflow.exceptions import UserErrorException
 
 
 def main():
+    command_args = sys.argv[1:]
+
+    parser = argparse.ArgumentParser(
+        prog="pfs",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description="Prompt Flow Service",
+    )
+
+    parser.add_argument("-p", "--port", type=int, help="port of the promptflow service")
+    args = parser.parse_args(command_args)
+    port = args.port
+
+    if port and is_port_in_use():
+        raise UserErrorException(f"Service port {port} is used.")
+    if not port:
+        # Read and write permission for user.
+        mode = stat.S_IRUSR | stat.S_IWUSR
```

Comment:
please reuse the logic in configuration part

Reply:
reuse this login in configuration